### PR TITLE
Stairs bugfix

### PR DESCRIPTION
**Pull request by AElanagai**
# Description

The bugs related to stairs are fixed here , Th first being the duplication of boss room every time 'B' key binder is clicked there was new instances of boss room created which Is solved in this PR and the Stairs now don't remain in normal rooms after player enters the boss room.

Fixes / Closes # (issue)

#433 
#432 

## Type of change

- Bug fix 

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by dhruvanil55 — COMMENTED** [review]
Can we also add the stuff from the other pr you're making into the team branch (the UI stuff) into this pr please

**Comment by vsinh14** [comment]
@AElanagai please resolve merge conflicts and the comments highlighted above

File: source/core/src/main/com/csse3200/game/areas/BossRoom.java [hunk 1]
```diff
@@ -57,20 +59,27 @@ public BossRoom(NPCFactory npcFactory,
         this.npcFactory = npcFactory;
     }
 
+    public boolean isBossRoom() {
```
[the remark this answers is not in the document]
Comment (AElanagai) [reply source/core/src/main/com/csse3200/game/areas/BossRoom.java]: yeah wasn't needed as technically its already   boss room class is extended from Enemy room

**Review by vsinh14 — APPROVED** [review]
All changes work as expected and all requested changes made

**Review by dhruvanil55 — APPROVED** [review]
LGTM

**Pull request merged**